packages/apps/dashboard/server/test/app.e2e-spec.ts

```diff
@@ -0,0 +1,96 @@
+import { Test, TestingModule } from '@nestjs/testing';
+import { INestApplication } from '@nestjs/common';
+import request from 'supertest';
+import { AppModule } from '../src/app.module';
+import { beforeAll, afterAll, describe, it, expect } from '@jest/globals';
+import { ConfigService } from '@nestjs/config';
+import {
+  TestEnvironmentConfigService,
+  testEnvValidator,
+} from '../src/common/config/test-environment-config.service';
+import { CACHE_MANAGER } from '@nestjs/cache-manager';
+import { Cache } from 'cache-manager';
+
+describe('Dashboard (e2e) tests', () => {
+  let app: INestApplication;
+  let configService: ConfigService;
+  let envConfigService: TestEnvironmentConfigService;
+  let cacheManager: Cache;
+
+  beforeAll(async () => {
+    const moduleFixture: TestingModule = await Test.createTestingModule({
+      imports: [AppModule],
+    }).compile();
+
+    app = moduleFixture.createNestApplication();
+    configService = moduleFixture.get<ConfigService>(ConfigService);
+    envConfigService = new TestEnvironmentConfigService(configService);
+    cacheManager = app.get<Cache>(CACHE_MANAGER);
+
+    const { error } = testEnvValidator.validate({
+      E2E_TESTING_EMAIL_ADDRESS: envConfigService.e2eTestingEmailAddress,
+    });
+
+    if (error) {
+      throw new Error(`Test environment is not valid: ${error.message}`);
+    }
+
+    await app.init();
+  });
+
+  describe('Networks API', () => {
+    it('should successfully retrieve operating networks and validate chain IDs', async () => {
+      const response = await request(app.getHttpServer())
+        .get('/networks/operating')
+        .expect(200);
+
+      expect(Array.isArray(response.body)).toBe(true);
+      expect(response.body.every((id: any) => typeof id === 'number')).toBe(
+        true,
+      );
+    });
+
+    it('should retrieve cached operating networks on subsequent requests', async () => {
+      const firstResponse = await request(app.getHttpServer())
+        .get('/networks/operating')
+        .expect(200);
+
+      const secondResponse = await request(app.getHttpServer())
+        .get('/networks/operating')
+        .expect(200);
+
+      expect(secondResponse.body).toEqual(firstResponse.body);
+    });
+
+    it('should set the cache when it does not exist', async () => {
+      await cacheManager.del('operating-networks');
+      jest.spyOn(cacheManager, 'set');
+
+      await request(app.getHttpServer()).get('/networks/operating').expect(200);
+
+      expect(cacheManager.set).toHaveBeenCalledWith(
+        'operating-networks',
+        expect.any(Array),
+        expect.any(Number),
+      );
+    });
+
+    it('should retrieve from cache when it exists', async () => {
+      const mockData = [1, 2, 3];
+      await cacheManager.set('operating-networks', mockData);
+
+      jest.spyOn(cacheManager, 'get');
```

Comment:
In e2e tests you don't need to spy, you only need to check it's actually working